### DISCUSSION TITLE
Streamline arithmetic aggregation across multiple columns

### DISCUSSION THREAD

**Discussion by imathews** [redivis/meta#141] [Feature requests]
**What is the problem you'd like us to address**
Currently, there's not a good way to sum laterally across multiple variables.

**Describe a solution (if you have one in mind)**
Either support a more robust arithmetic input or expose a SUM() method that takes multiple variables as an input.

**Comment by sfmcintyre** [comment]
Consider logical sum in addition to arithmetic sum, and potentially configurable behavior for NULL values

Additional comments by Josh Grab in PHS Users slack:
> Feature request for transforms:  I often need to create a variable that is a simple sum of more than two variables.  To sum two variables, I use the Arithmetic method, which gives boxes for exactly two input variables and an operator.  If I want to sum K variables (where K>2), that is cumbersome, because then I need K-1 Arithmetic blocks and K-2 intermediate variables (well, if I combine pairwise, I could get by with fewer steps, but it would still be cumbersome) or I need to write a custom SQL query to do it, which has other issues I won't go into here.

> [Note:  There is a "Sum" method under Create variables, but that is "Aggregate --> Sum" which sums across rows by groups-- I want to sum across columns.]

> I'd envision this being an option listed near the "Arithmetic" option, perhaps called "Arithmetic Sum Across Columns" or something like that, with one input field for listing all the fields to use in the sum, and perhaps a check box for whether to return NULL if any of the addends are NULL (with default behavior being "compute the sum of all non-NULL variables listed, returning NULL if all inputs are NULL").

> P.S.  It might be nice to extend or add a similar method that would "add" Boolean variables (return TRUE if any argument is TRUE, with checkbox to toggle to "if all arguments are TRUE").  In other words, column-wise "Logical OR" function and "Logical AND" function.  Not sure if I'd put those under "Arithmetic".  Probably I'd add a "Logic" method with options "Logical AND (of columns)", "Logical OR (of columns)".
